src/Data/DataResponse.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Data;
+
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * A wrapper around PSR-7 response that is assigned raw data to be formatted later using a formatter.
+ *
+ * For example, `['name' => 'Dmitriy']` to be formatted to JSON using {@see \Yiisoft\Yii\Web\Data\Formatter\JsonDataResponseFormatter}
+ * when {@see DataResponse::getBody()} is called.
+ */
+class DataResponse implements ResponseInterface
+{
+    private ResponseInterface $response;
+
+    private $data;
+
+    private ?StreamInterface $dataStream = null;
+
+    private ?DataResponseFormatterInterface $responseFormatter = null;
+
+    public function __construct($data, int $code, string $reasonPhrase, ResponseFactoryInterface $responseFactory)
```

Comment:
The `ResponseFactory` used only for creating a `Response`.
Maybe need to pass the `Response` instead of `ResponseFactory`?

Reply:
ab80160774bd23975152bf0bf5d397e6c9e060e7